homeassistant/components/switchbee/climate.py

```diff
@@ -0,0 +1,218 @@
+from homeassistant.components.climate.const import (
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    ClimateEntityFeature,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS, TEMP_FAHRENHEIT
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import SwitchBeeCoordinator
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+FAN_SB_TO_HASS = {
+    ThermostatFanSpeed.AUTO: FAN_AUTO,
+    ThermostatFanSpeed.LOW: FAN_LOW,
+    ThermostatFanSpeed.MEDIUM: FAN_MEDIUM,
+    ThermostatFanSpeed.HIGH: FAN_HIGH,
+}
+
+FAN_HASS_TO_SB = {
+    FAN_AUTO: ThermostatFanSpeed.AUTO,
+    FAN_LOW: ThermostatFanSpeed.LOW,
+    FAN_MEDIUM: ThermostatFanSpeed.MEDIUM,
+    FAN_HIGH: ThermostatFanSpeed.HIGH,
+}
+
+HVAC_MODE_SB_TO_HASS = {
+    ThermostatMode.COOL: HVACMode.COOL,
+    ThermostatMode.HEAT: HVACMode.HEAT,
+    ThermostatMode.FAN: HVACMode.FAN_ONLY,
+}
+
+HVAC_MODE_HASS_TO_SB = {
+    HVACMode.COOL: ThermostatMode.COOL,
+    HVACMode.HEAT: ThermostatMode.HEAT,
+    HVACMode.FAN_ONLY: ThermostatMode.FAN,
+}
+
+HVAC_ACTION_SB_TO_HASS = {
+    ThermostatMode.COOL: HVACAction.COOLING,
+    ThermostatMode.HEAT: HVACAction.HEATING,
+    ThermostatMode.FAN: HVACAction.FAN,
+}
+
+HVAC_UNIT_SB_TO_HASS = {
+    ThermostatTemperatureUnit.CELSIUS: TEMP_CELSIUS,
+    ThermostatTemperatureUnit.FAHRENHEIT: TEMP_FAHRENHEIT,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Switchbee thermostat."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        Device(hass, device, coordinator)
+        for device in coordinator.data.values()
+        if device.type == DeviceType.Thermostat
+    )
+
+
+class Device(CoordinatorEntity, ClimateEntity):
+    """Representation of an Switchbee button."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        device: SwitchBeeThermostat,
+        coordinator: SwitchBeeCoordinator,
+    ) -> None:
+        """Initialize the Switchbee switch."""
+        super().__init__(coordinator)
+        self._session = aiohttp_client.async_get_clientsession(hass)
+        self._attr_name = f"{device.zone} {device.name}"
+        self._device_id = device.id
+        self._attr_unique_id = f"{coordinator.mac_formated}-{device.id}"
+        self._attr_supported_features = (
+            ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.FAN_MODE
+        )
+
+        # set HVAC capabilities
+        self._attr_fan_modes = [FAN_AUTO, FAN_HIGH, FAN_MEDIUM, FAN_LOW]
```

Comment:
If you set this at the class level you will avoid a new list being created for each instance of a device.

You could really do this way any constant value that doesn't depend on the device, but in Python the int and string values will point to global instances rather than create new ones anyway, so it won't make a difference there.

Reply:
Thanks @ViViDboarder, how do you suggest handling this?